imu.c

```diff
@@ -0,0 +1,464 @@
+         * are averaged, resulting in 4 averaged samples.
+         * Note1 : For more information, refer the datasheet.
+         * Note2 : A higher number of averaged samples will result in a lower noise level of the signal, but
+         * this has an adverse effect on the power consumed.
+         */
+        config[ACCEL].cfg.acc.bwp = BMI2_ACC_NORMAL_AVG4;
+
+        /* Enable the filter performance mode where averaging of samples
+         * will be done based on above set bandwidth and ODR.
+         * There are two modes
+         *  0 -> Ultra low power mode
+         *  1 -> High performance mode(Default)
+         * For more info refer datasheet.
+         */
+        config[ACCEL].cfg.acc.filter_perf = BMI2_PERF_OPT_MODE;
+
+        /* The user can change the following configuration parameters according to their requirement. */
+        /* Set Output Data Rate */
+        config[GYRO].cfg.gyr.odr = BMI2_GYR_ODR_200HZ;
+
+        /* Gyroscope Angular Rate Measurement Range.By default the range is 2000dps. */
+        config[GYRO].cfg.gyr.range = BMI2_GYR_RANGE_2000;
+
+        /* Gyroscope bandwidth parameters. By default the gyro bandwidth is in normal mode. */
+        config[GYRO].cfg.gyr.bwp = BMI2_GYR_NORMAL_MODE;
+
+        /* Enable/Disable the noise performance mode for precision yaw rate sensing
+         * There are two modes
+         *  0 -> Ultra low power mode(Default)
+         *  1 -> High performance mode
+         */
+        config[GYRO].cfg.gyr.noise_perf = BMI2_POWER_OPT_MODE;
+
+        /* Enable/Disable the filter performance mode where averaging of samples
+         * will be done based on above set bandwidth and ODR.
+         * There are two modes
+         *  0 -> Ultra low power mode
+         *  1 -> High performance mode(Default)
+         */
+        config[GYRO].cfg.gyr.filter_perf = BMI2_PERF_OPT_MODE;
+
+        /* Set the accel and gyro configurations. */
+        rslt = bmi2_set_sensor_config(config, 2, bmi);
+        bmi2_print_error_code(rslt);
+    }
+
+    return rslt;
+}
+
+/******************************************************************************/
+
+/* Status of api are returned to this variable. */
+static int8_t rslt;
+
+/* Variable to define limit to print accel data. */
+static uint8_t limit = 10;
+
+/* Assign accel sensor to variable. */
+static uint8_t sensor_list[2] = {BMI2_ACCEL, BMI2_GYRO};
+
+// Primary IMU device structure; interface interface is done here
+// replaces bmi2_interface_init
+static struct bmi2_dev bmi = {
+     // Set SPI interface parameters
+     .chip_id = BMI270_CHIP_ID, 
+     .dummy_byte = 1,
+     .read_write_len = 8, 
+     .intf = BMI2_SPI_INTF,
+     .read = imuRead,
+     .write = imuWrite,
+     .delay_us = imuDelay,
+     .config_file_ptr = NULL
+};
+
+/* Structure to define type of sensor and their respective data. */
+static struct bmi2_sens_data sensor_data = { { 0 } };
+
+static uint8_t indx = 0;
+float accX = 0, accY = 0, accZ = 0;
```

Comment:
static on all of these too, also static variables at the top of the file, then static functions.

everything from 282 - 313 should be at the top